### PR TITLE
Add declarations for jquery 1,2,3 and react 15, 16; Add smoke tests

### PR DESCRIPTION
**Pull request by bashor**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -166,6 +166,86 @@ subprojects.stream().filter { it.parent != rootProject }.each { p ->
...
+                }
+
+                def moduleKind = it.name.substring(start, end).toLowerCase()
+
+                if (moduleKind.isEmpty()) {
+                    moduleKind = "umd"
+                }
+
+                it.kotlinOptions.moduleKind = moduleKind
+                test.dependsOn(it)
```
Comment (anton-bannykh) [inline build.gradle:214]: IMHO a better solution would be specifying a special property in konfig.json, which tells us which module systems the library is supposed to work with, e.g. `moduleSystems: ['plain', 'commonjs', 'umd']`. I thinks that's much easier to explain and configure.

File: build.gradle [hunk 2]
```diff
@@ -166,6 +166,86 @@ subprojects.stream().filter { it.parent != rootProject }.each { p ->
...
+
+            testSharedPlainOnly {
+                kotlin.srcDirs += "$projectDir/../_testSharedPlain"
+                compileClasspath = test.compileClasspath
+            }
+
+            testSharedCommonJsOnly {
+                kotlin.srcDirs += "$projectDir/../_testSharedCommonJS"
+                compileClasspath = test.compileClasspath
+            }
```
Comment (anton-bannykh) [inline build.gradle]: Why not just add "$projectDir/../_testSharedCommonJS" to testSharedCommonJS?

File: externals/react/v15/src/index.React.kt [hunk 3]
```diff
@@ -0,0 +1,1784 @@
+@file:Suppress("INTERFACE_WITH_SUPERCLASS", "OVERRIDING_FINAL_MEMBER", "RETURN_TYPE_MISMATCH_ON_OVERRIDE", "CONFLICTING_OVERLOADS", "EXTERNAL_DELEGATION", "NESTED_CLASS_IN_EXTERNAL_INTERFACE")
+@file:JsQualifier("React")
+package React
```
Comment (anton-bannykh) [inline externals/react/v15/src/index.React.kt]: Wrong package?

**Review by anton-bannykh — CHANGES_REQUESTED** [review]
Tests-related comments are minor
Packages are pretty important I think

File: externals/react/v15/src/index.React.kt (hunk 3 above)
Comment (bashor) [reply externals/react/v15/src/index.React.kt]: Fixed

File: build.gradle (hunk 1 above)
Comment (bashor) [reply build.gradle:214]: I think both solutions have advantages and disadvantages.
I propose to try to live with this solution now. If we will not like it we will switch to another.

File: build.gradle (hunk 2 above)
Comment (bashor) [reply build.gradle]: Done

File: build.gradle (hunk 1 above)
Comment (anton-bannykh) [reply build.gradle:214]: Ok)

**Review by anton-bannykh — APPROVED** [review]

**Comment by anton-bannykh** [comment]
lgtm

**Comment by bashor** [comment]
Merged manually

**Pull request closed without merging**